### ISSUE TITLE
Add multi-GPU training example (DDP)

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#52]
## Summary
Create a multi-GPU training example using DDP (DistributedDataParallel) or FSDP. This validates that Opaque works in distributed settings.

## Current State
- `adaptive_clipped_grad` docstring has a DDP example snippet
- `PoissonSampler` has distributed support (PRs #14, #17, #20)
- No runnable multi-GPU example

## Implementation
Two approaches (DDP is simpler, FSDP2 is more powerful):

### DDP Approach (start here)
- Wrap model in `torch.nn.parallel.DistributedDataParallel`
- Each rank runs `clipped_grad()` independently on its local batch
- All-reduce summed gradients across ranks
- Add noise on rank 0, broadcast to all ranks
- Accounting: single `Accountant` on rank 0

### FSDP2 Approach (future, depends on #25)
- Full parameter sharding across ranks
- Requires #55-#58 to be resolved first

## Script Structure
```python
# examples/train_causal_lm_distributed.py
# or: train_causal_lm.py --distributed ddp --nproc 4

torchrun --nproc_per_node=4 examples/train_causal_lm.py --distributed
```

## Acceptance Criteria
- [ ] DDP example works with ≥ 2 GPUs
- [ ] Privacy accounting is correct (accounts for full batch across all ranks)
- [ ] Noise added once (not per-rank)
- [ ] Results match single-GPU training (within noise)
- [ ] Launch script documented (`torchrun` command)

## Depends on
- #47 (accounting in training script)

## Parent
- #65
